### DISCUSSION TITLE
Reverse Proxy - any security issues?

### DISCUSSION THREAD

**Discussion by Geffers** [pivpn/pivpn#1867] [Support]
Folks, used a PiVPN on my local network for a couple of years now, great for accessing remotely my home servers, no need to open ports.  Also been great when abroad accessing UK TV services.

Owing to a change of Internet provider who is now behind a CGNAT I've had to explore other options, temporarily I have managed to get a reverse SSH proxy to work,  I have a cloud ubuntu system working via Oracle, by connecting via ssh from my PiVPN server to my cloud Ubuntu and using -R option I am able to log on to my PiVPN but using my cloud IP address.

Now my logic suggests all is encrypted and there are no security holes in this type of connection but appreciate comments if there are any potential issues.

I did have to use this via TCP rather than UDP and presently not sure of any repercussions, I don't use the PiVPN extensively.

Geoff

**Comment by orazioedoardo** [comment]
So you are doing RaspberryPi -> Oracle Ubuntu <- Computer? I think this way technically the VM can access the transmitted data.

**Reply by Geffers** [reply]
Almost, it is ssh with a reverse proxy so connection is;

PiVPn -> Oracale Ubuntu -> PiVPN    so encrypted ssh connection forwarding just port 1194

Client VPN then connects to Oracle and guessing from there the original SSH tunnel then encrypts the data back to the PiVPN to make the connection.

Temporary measure as I shall have to go to IPv6 as this IPv4 behind a CGNAT system has upended my Virtual LAN set up plus extra SSID to keep IoT devices away from personal devices.

Geoff
